src/DotLiquid/StandardFilters.cs

```diff
@@ -827,10 +872,25 @@ public static IEnumerable Uniq(object input)
         /// </summary>
         /// <param name="context">The DotLiquid context</param>
         /// <param name="input">Input to be transformed by this filter</param>
-        public static double Abs(Context context, object input)
+        [LiquidFilter(MinVersion = SyntaxCompatibility.DotLiquid24)]
+        public static object Abs(Context context, object input)
         {
-            Double n;
-            return Double.TryParse(input.ToString(), NumberStyles.Number, context.CurrentCulture, out n) ? Math.Abs(n) : 0;
+            if (input == null) return 0;
+
+            if (input is string inputString)
+            {
+                input = NumericConverter.CoerceToNumericType(inputString, context.FormatProvider, 0);
+            }
+
+            if (input is decimal inputDecimal) { return Math.Abs(inputDecimal); }
+            else if (input is float inputFloat) { return Math.Abs(inputFloat); }
+            else if (input is double inputDouble) { return Math.Abs(inputDouble); }
+            else if (input is sbyte inputInt8) { return Math.Abs(inputInt8); }

```

Comment:
C# is unable to handle sbyte.MinValue as the result exceeds the MaxValue.

```c#
public static object Abs(object input) {
	try 
	{
		var decInput = Convert.ToDecimal(input);
		var decResult = Math.Abs(decInput);
		
		try {
			return Convert.ChangeType(decResult, input.GetType());
		} catch (OverflowException) 
		{
			return Convert.ChangeType(decResult, NumericTypePromotions[input.GetType()][0]);
		}
	} 
	catch (OverflowException) 
	{
		var dblInput = Convert.ToDouble(input);
		var dblResult = Math.Abs(dblInput);
		return Convert.ChangeType(dblResult, input.GetType());
	}
}
```

Attached updated 
[POCAtLeastNumerics.zip](https://github.com/user-attachments/files/19597783/POCAtLeastNumerics.zip)